Sources/Knit/Module/DependencyBuilder.swift

```diff
@@ -212,3 +206,37 @@ extension ModuleAssembly {
     }
 
 }
+
+internal extension DependencyBuilder {
+
+    struct RegistrationCache {
```

Comment:
This cache allows more checks use .contains rather than iterating arrays for matches.

Reply:
Nice 👍